future-apps/payroll/test/contracts/Payroll.test.js

```diff
@@ -0,0 +1,3362 @@
+const { assertRevert } = require('@aragon/test-helpers/assertThrow')
+const { encodeCallScript } = require('@aragon/test-helpers/evmScript')
+const { getEvent, getEvents, getEventArgument } = require('../helpers/events')
+const { bn, bigExp, maxUint64, maxUint256, annualSalary } = require('../helpers/numbers')(web3)
+const { deployErc20TokenAndDeposit, redistributeEth, getDaoFinanceVault } = require('../helpers/setup.js')(artifacts, web3)
+
+const ACL = artifacts.require('ACL')
+const Payroll = artifacts.require('PayrollMock')
+const PriceFeed = artifacts.require('PriceFeedMock')
+const ExecutionTarget = artifacts.require('ExecutionTarget')
+
+const MAX_GAS_USED = 6.5e6
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+contract('Payroll, initialization,', ([owner, employee, anotherEmployee, anyone]) => {
+  let dao, acl, payroll, payrollBase, finance, vault, priceFeed, denominationToken, anotherToken
+
+  const NOW = 1553703809 // random fixed timestamp in seconds
+  const ONE_MONTH = 60 * 60 * 24 * 31
+  const TWO_MONTHS = ONE_MONTH * 2
+  const RATE_EXPIRATION_TIME = TWO_MONTHS
```

Comment:
This is a bit unrealistic, since it's so long. Perhaps we could use something like thirty minutes instead?

Reply:
I chose this value to make it simpler to deal with accruing salaries for the tests, but I also tested many different scenarios of expired rates :)